security/vault.go

```diff
@@ -0,0 +1,242 @@
+package security
+
+import (
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/gaia-pipeline/gaia"
+)
+
+const (
+	vaultName = ".gaia_vault"
+)
+
+// Vault is a secret storage for data that gaia needs to store encrypted.
+type Vault struct {
+	Path string
```

Comment:
Is it on purpose that Path and Cert are global available?

Reply:
You mean exported? Hum, haven't really thought about it, but you are right. They aren't used directly, so they can be private for only internal use. Thanks. :)